elastic/kibana
# [Reporting] Baseline capture tests

## Summary

This contribution expands the baseline report capturing testing we have. Specifically it adds:

* one place where the different report settings are all tested (PNG, PDF, PDF print optimized)
* Adds tests for comparing PDF and PDF print-optimized against a baseline
* Introduces a single source of image comparison logic in the `compareImages` test service (if we merge this PR we should remove the `checkIfPngsMatch` logic from dashboard).

## How to test

1. Inside the `x-pack` directory run `node ../scripts/functional_tests_server.js --config ./test/examples/config.ts`
2. Still inside `x-pack`, once the test server is ready, run ` node ./scripts/functional_test_runner.js --config ./test/examples/config.ts --grep 'Capture'`
3. Check the test output


### Risk Matrix

Before closing this PR, invite QA, stakeholders, and other developers to identify risks that should be tested prior to the change/feature release.

When forming the risk matrix, consider some of the following examples and how they may potentially impact the change:

| Risk                      | Probability | Severity | Mitigation/Notes        |
|---------------------------|-------------|----------|-------------------------|
| New source of test flakiness | Med | Low | These tests act as an additional guard for making sure that all of our reporting functionality is tested behind automation. Further mitigations against potential flakiness will require further investigation. |
